### ISSUE TITLE
Test case: Bug 1459352 - Dark theme warnings

### ISSUE THREAD

**Issue by tracywalker** [mozilla/iris#1825]
https://testrail.stage.mozaws.net/index.php?/cases/edit/171347

**Comment by icrisan** [comment]
I don't think that we can automate this kind of test cases. @tracywalker  if you think so also please close this issue accordingly. Thank you!

**Comment by tracywalker** [comment]
Ioana, what part of the test are you concerned with not being able to automate?  The theme change is in Firefox.  It is not a system theme change to dark.  We should be able to automate this.

**Comment by icrisan** [comment]
Yes, we can change the theme in Firefox. We already have tests that are doing that like "key_navigation_high_contrast_theme.py" test case. My concern is the validation of the expected results from step 4. I think that a human eye can validate better if the message is readable(the gray color used is good enough).

**Comment by tracywalker** [comment]
Well of course, human eye can detect most everything better than Iris.  Do you think Iris can reliably perform the final verification?

**Comment by icrisan** [comment]
If we use a similarity big enough yes otherwise no. If a 0.95 similarity works on all the OSes/machines I think that it would be reliable to test this with Iris. If we have to adjust the similarity so the test case works on my side or your side I think that maybe we should not automate this test case.

**Comment by tracywalker** [comment]
For now, automate the test case. If it needs 0.99, we can then disable it until the core bug (issue #2920) on exact() is fixed.

**Comment by icrisan** [comment]
Tried to automate this test case and got the following while running test case with DEBUG on my retina MAC:
2019-04-23 12:47:53 [DEBUG] [iris.api.core.util.image_search] Searching for pattern: warning_message_tracking_protection.png
2019-04-23 12:47:53 [DEBUG] [iris.api.core.util.image_search] Match score: 0.928444862366. Desired precision: 0.95

So, the pattern that I took from my retina MAC is found with only 0.928444862366 precision. Maybe on your MAC it is found with a precision even smaller. I can decrease the similarity to 0.8, the test case will PASS,  but I in this case I cannot be sure if the gray color used is the correct one.
